lib/connection-pool.js

```diff
@@ -220,6 +221,7 @@ function setFree(pooled) {
     pooled.status = FREE;
     pooled.timeout = setTimeout(function() {
         self.log('closing idle connection: ' + pooled.id);
+        pooled.status = PENDING;
```

Comment:
This isn't necessary. The `end` event handler is called synchronously in the `close()` function. The connection status will be set to RETRY by that `end` event handler before any other runs.